tests/rbd_mirror/test_rbd_mirror_image_features.py

```diff
@@ -0,0 +1,211 @@
+        imagespec = poolname + "/" + imagename
+
+        image_feature = "object-map,fast-diff"
+        features = list(image_feature.split(","))
+        # Disable image features in primary cluster
+        mirror1.image_feature_disable(imagespec=imagespec, image_feature=image_feature)
+        # wait till image gets reflected to secondary
+        time.sleep(60)
+        # Verify image feature reflection in secondary cluster
+        out = rbd2.image_info(poolname, imagename)
+        for word in features:
+            if word in out["features"]:
+                log.error("Image features not disabled in secondary")
+                return 1
+        log.info("Image features got disabled successfully in secondary")
+
+        # Enable image features in primary cluster
+        mirror1.image_feature_enable(imagespec=imagespec, image_feature=image_feature)
+        time.sleep(60)
+        # Verify image feature reflection in secondary cluster
+        out = rbd2.image_info(poolname, imagename)
+        for word in features:
+            if word not in out["features"]:
+                log.error("Image features not enabled in secondary site")
+                return 1
+        log.info("Image features got enabled successfully in secondary")
+        return 0
+
+    except ValueError as ve:
+        log.error(
+            f"{kw.get('ceph_cluster_dict').values} has less or more clusters Than Expected(2 clusters expected)"
+        )
+        log.exception(ve)
+        return 1
+    except Exception as e:
+        log.exception(e)
+        return 1
+    finally:
+        mirror1.clean_up(peercluster=mirror2, pools=[poolname])
+
+
+def test_snapshot_image_feature(rbd_mirror, pool_type, **kw):
+    """Method to validate change of image feature in primary
+    got reflected onto secondary in snapshot based mirroring.
+    Args:
+        rbd_mirror: Object
+        pool_type: Replicated or EC pool
+        **kw: test data
+    Returns:
+        int: The return value. 0 for success, 1 otherwise
+    """
+
+    try:
+        log.info(
+            "Running RBD snapshot based mirroring test case for image features reflection"
+        )
+        config = kw.get("config")
+        mirror1 = rbd_mirror.get("mirror1")
+        mirror2 = rbd_mirror.get("mirror2")
+        rbd1, rbd2 = [
+            Rbd(**kw, req_cname=cluster_name)
+            for cluster_name in kw.get("ceph_cluster_dict").keys()
+        ]
+        poolname = config[pool_type]["pool"]
+        imagename = config[pool_type]["image"]
+        imagespec = poolname + "/" + imagename
+
+        image_feature = "object-map,fast-diff"
+        search_pattern = ["object map invalid", "fast diff invalid"]
+        # Disable image features in primary cluster
+        mirror1.image_feature_disable(imagespec=imagespec, image_feature=image_feature)
+        mirror2.resync(imagespec=imagespec)
+        # wait till image gets reflected to secondary
+        time.sleep(60)
```

Comment:
use iterator to check the image features availability

Reply:
done